src/ir.cpp

```diff
@@ -18747,6 +18764,13 @@ static TypeTableEntry *ir_analyze_instruction_frame_address(IrAnalyze *ira, IrIn
     return u8_ptr_const;
 }
 
+static TypeTableEntry *ir_analyze_instruction_handle(IrAnalyze *ira, IrInstructionHandle *instruction) {
+    ir_build_handle_from(&ira->new_irb, &instruction->base);
+
+    TypeTableEntry *promise_type = get_promise_type(ira->codegen, nullptr);
```

Comment:
Here, I think that `@handle()` should return a `promise->T` rather than just a `promise`, because we know the return type of the async function. If you look at the type of the variable that suspend blocks provide, it does this. You can see an example of this in `ir_analyze_instruction_coro_begin`.